mypy/messages.py

```diff
@@ -634,8 +634,18 @@ def incompatible_argument(self, n: int, m: int, callee: CallableType, arg_type:
                 arg_type_str = '*' + arg_type_str
             elif arg_kind == ARG_STAR2:
                 arg_type_str = '**' + arg_type_str
-            msg = 'Argument {} {}has incompatible type {}; expected {}'.format(
-                n, target, self.quote_type_string(arg_type_str),
+
+            # For function calls with keyword arguments, use the argument name instead of the
+            # number.
+            arg_name = context.arg_names[n - 1] if isinstance(context, CallExpr) else None
+            if arg_name:
+                message_format = 'Argument "{}" {}has incompatible type {}; expected {}'
```

Comment:
I think it would be cleaner to make the computation of `arg_name` conditional (and either quoted or not) while leaving the main format string unconditional, to reduce duplication.